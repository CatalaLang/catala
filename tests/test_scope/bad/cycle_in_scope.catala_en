## Article

```catala
declaration scope A:
  context x content integer
  context y content integer
  context z content integer

scope A:
  definition y under condition x < 0 consequence equals - x
  definition y under condition x >= 0 consequence equals x
  definition z under condition y >= 1 consequence equals 10 / y
  definition z under condition y < 1 consequence equals y
  definition x equals z
```
