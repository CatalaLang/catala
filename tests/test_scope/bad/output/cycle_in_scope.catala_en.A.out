[ERROR] Cyclic dependency detected between variables of scope A!
[ERROR] 
[ERROR] Cycle variable z, declared:
[ERROR]  --> test_scope/bad/cycle_in_scope.catala_en
[ERROR]   | 
[ERROR] 7 |   context z content integer
[ERROR]   |           ^
[ERROR]   + Article
[ERROR] 
[ERROR] Used here in the definition of another cycle variable x:
[ERROR]   --> test_scope/bad/cycle_in_scope.catala_en
[ERROR]    | 
[ERROR] 14 |   definition x equals z
[ERROR]    |                       ^
[ERROR]    + Article
[ERROR] 
[ERROR] Cycle variable y, declared:
[ERROR]  --> test_scope/bad/cycle_in_scope.catala_en
[ERROR]   | 
[ERROR] 6 |   context y content integer
[ERROR]   |           ^
[ERROR]   + Article
[ERROR] 
[ERROR] Used here in the definition of another cycle variable z:
[ERROR]   --> test_scope/bad/cycle_in_scope.catala_en
[ERROR]    | 
[ERROR] 13 |   definition z under condition y < 1 consequence equals y
[ERROR]    |                                ^
[ERROR]    + Article
[ERROR] 
[ERROR] Cycle variable x, declared:
[ERROR]  --> test_scope/bad/cycle_in_scope.catala_en
[ERROR]   | 
[ERROR] 5 |   context x content integer
[ERROR]   |           ^
[ERROR]   + Article
[ERROR] 
[ERROR] Used here in the definition of another cycle variable y:
[ERROR]   --> test_scope/bad/cycle_in_scope.catala_en
[ERROR]    | 
[ERROR] 11 |   definition y under condition x >= 0 consequence equals x
[ERROR]    |                                ^
[ERROR]    + Article
